### PR TITLE
completes Task 5.GET/api/articles/:article_id

### PR DESCRIPTION
**Pull request by Rawan-Hamza**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -107,3 +107,36 @@ describe("* GET/api/articles", () => {
...
+      expect(body).toEqual(
+        {
+          article_id: 3,
+          title: 'Eight pug gifs that remind me of mitch',
+            topic: 'mitch',
+            author: 'icellusedkars',
+            body: 'some gifs',
+            created_at: '2020-11-03T09:12:00.000Z',
+            votes: 0
+        }
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: We want to make this test less "brittle" so the article is open to extension (e.g. if another property is added to the article data in future). To do this we need to move away from `toEqual` and rely on flexible matches like `toMatchObject` or `expect.objectContaining` which will match for properties but won't fail if more properties are present in future.

I'd refactor this test to be more flexible in that sense

File: __tests__/app.test.js [hunk 2]
```diff
@@ -107,3 +107,36 @@ describe("* GET/api/articles", () => {
...
+      )
+    });
+  });
+      
+  test("returns status code 400 when given invalid article_id", () => {
+        return request(app).get("/api/articles/1652342756").expect(400)
+            .then((response) => {
+              const msg = response.body.msg;
+              expect(msg).toBe("bad request")
+            })
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: This test wouldn't actually be a '400 - Bad Request' in my opinion. The user technically isn't doing anything wrong, it's just that article id does not exist on the database. This is actually a 404 Not Found error, please amend to reflect that

however we do need to account for Bad Requests e.g. '/api/articles/notAnId' which is a bad user behaviour, please add another test to account for that scenario

File: __tests__/app.test.js [hunk 3]
```diff
@@ -107,3 +107,36 @@ describe("* GET/api/articles", () => {
...
+
+describe("* GET/api/articles/:article_id", () => {
+  test("returns status code 200", () => {
+    return request(app).get("/api/articles/3").expect(200);
+  });
+
+  test("responds with a single matching article", () => {
+    return request(app).get("/api/articles/3").expect(200)
+    .then(({ body }) => {
+      expect(body).toEqual(
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: You should be sending data back on the key of 'article' e.g.
```
{ body: article : { article_id: 3 .. } } 
```

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
An additional test needed and an amendment to the current sad path in this test suite. Very nice solution code so far - makes so much sense 🚀

File: __tests__/app.test.js (hunk 1 above)
Comment (Rawan-Hamza) [reply __tests__/app.test.js]: this has been refactored to match previous tests. thank you

File: __tests__/app.test.js (hunk 2 above)
Comment (Rawan-Hamza) [reply __tests__/app.test.js]: That's great I was actually stuck on this yesterday, but now it should have both tests

File: __tests__/app.test.js (hunk 3 above)
Comment (Rawan-Hamza) [reply __tests__/app.test.js]: this has been refactored, thank you :)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -114,29 +114,41 @@ describe("* GET/api/articles/:article_id", () => {
...
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+          })
+        );
+      });
```
Comment (dav3rid) [inline __tests__/app.test.js:132]: I think there may have been some confusion regarding Poonam's last comments...
You were totally correct to assert for specific values as we are requesting a specific article by its ID.
What Poonam was suggesting was to wrap that in a `expect.objectContaining` just to allow for further properties to be present in future

**Review by dav3rid — APPROVED** [review]

**Pull request merged**